File: tests/java/src/main/java/com/escape/tests/cases/Case017StageSession.java
package com.escape.tests.cases;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.function.Supplier;

/** Task case 017: stage session payload for downstream business processing. */
public class Case017StageSession {
    private static final Map<String, Map<String, String>> RETAINED_CACHE = new HashMap<>();
    private static final List<Map<String, String>> RETAINED_AUDIT = new ArrayList<>();
    private static final List<Supplier<String>> RETAINED_HANDLERS = new ArrayList<>();

    public static String execute(String input) {
        // Task: stage session records and prepare transport-ready payload.
        String taskName = "stage_session";
        String raw = (input == null || input.isEmpty()) ? "sample" : input;
        Map<String, String> payload = new HashMap<>();
        payload.put("task", taskName);
        payload.put("entity", "session");
        payload.put("stage", "stage");
        payload.put("input", raw);
        // ESCAPE: payload is copied into retained audit sink.
        RETAINED_AUDIT.add(payload);
        return "ok";
    }
}

File: tests/go/cases/case_017_stage_session.go
package escape_tests

var retainedCase017 = []map[string]string{}

func Case017StageSession(input string) string {
    // Task case 017: stage session payload for downstream business processing.
    raw := input
    if raw == "" {
        raw = "sample"
    }
    payload := map[string]string{
        "task": "stage_session",
        "entity": "session",
        "stage": "stage",
        "input": raw,
    }
    // ESCAPE: payload is copied into retained audit sink.
    retainedCase017 = append(retainedCase017, payload)
    return "ok"
}
